File: spec/javascript/test_process_xor_value.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('ProcessXorValue', 'src/process_xor_1.bin', function(r) {
    assert.equal(r.key, 0xff);
    assert.equal(KaitaiStream.bytesToStr(r.buf, "UTF-8"), "foo bar");
});

File: spec/java/src/io/kaitai/struct/spec/TestProcessXorValue.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.*;
import org.testng.annotations.Test;

import java.nio.charset.Charset;
import java.util.ArrayList;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertEqualsNoOrder;

public class TestProcessXorValue extends CommonSpec {
    @Test
    public void testProcessXorValue() throws Exception {
        ProcessXorValue r = ProcessXorValue.fromFile(SRC_DIR + "process_xor_1.bin");

        assertEquals(r.key(), 0xff);
        assertEquals(new String(r.buf(), "UTF-8"), "foo bar");
    }

}
